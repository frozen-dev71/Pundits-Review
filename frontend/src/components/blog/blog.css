/*
// ======================================
// CSS for Blog Home Component 
// ======================================
*/


.blog-page-wrapper {
	margin-top: 10vh;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}


.blog-page-title {
	font-size: 1.5em;
	letter-spacing: .1em;
	margin-bottom: 2em;
}

.blog-page-description {
	font-size: 1.25em;
    letter-spacing: .2em;
}



/*
// ======================================
// MEDIA QUERIES
// ======================================
*/



@media (min-width: 550px) {

	.blog-page-title {
		
		font-size: 2em;

	}

	.blog-page-description {
		font-size: 1.5em;
	    
	}
}



@media (min-width: 1000px) {

	.blog-page-title {
		
		font-size: 2.5em;

	}

	.blog-page-description {
		font-size: 2em;
	    
	}
}