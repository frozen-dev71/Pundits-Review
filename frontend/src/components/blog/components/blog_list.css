/*
// ======================================
// CSS for BLOG LIST Component 
// ======================================
*/



.blog-list-loading-container {
	display: flex;
	height: 50vh;
	width: 100vw;
	justify-content: center;
	align-items: center;
}


.blog-list-wrapper {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

