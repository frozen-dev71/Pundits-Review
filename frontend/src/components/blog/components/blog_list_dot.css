
/*
// ======================================
// CSS for BLOG List  dot
// ======================================
*/


.blog-list-dot {
	height: 5px;
    width: 5px;
    border-radius: 2em;
    background-color: black;
}