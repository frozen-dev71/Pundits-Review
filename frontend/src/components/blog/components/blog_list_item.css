/*
// ======================================
// CSS for Blog List Item
// ======================================
*/


.blog-list-item-wrapper {
	
	display: flex;
	flex-direction: column;
	cursor: pointer;
	align-items: center;
	justify-content: center;
	padding: 2em 0;
	max-width: 15em;

}

.blog-list-item-image {
	height: 10em;
    width: 15em;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    border: solid black 2px;
}


.blog-list-item-headline {
	font-size: 1em;
	letter-spacing: .2em;
	text-align: center;
	padding-bottom: 1em;
	color: black;
}

.blog-list-item-date {
	font-size: .65em;
	letter-spacing: .2em;
	padding: 1em 0;
	color: black;
}

.blog-list-item-description {
	font-size: .65em;
	letter-spacing: .1em;
	padding: 1em 0;
	text-align: center;
	color: black;
}




/*
// ======================================
// MEDIA QUERIES
// ======================================
*/



@media (min-width: 550px) {

	.blog-list-item-wrapper {
	
		padding: 3em 0;
		max-width: 20em;

	}

	.blog-list-item-image {
		height: 12em;
	    width: 20em;
	}


	.blog-list-item-headline {
		font-size: 1.25em;
	}

	.blog-list-item-date, .blog-list-item-description {
		font-size: .85em;
		padding: 1.75em 0;
	}

}


@media (min-width: 1200px) {

	.blog-list-item-wrapper {
	
		padding: 4em 0;
		max-width: 30em;

	}

	.blog-list-item-image {
		height: 20em;
	    width: 30em;
	}


	.blog-list-item-headline {
		font-size: 1.75em;
	}



}