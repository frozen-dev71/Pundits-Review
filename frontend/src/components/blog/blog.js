// ======================================
// Blog Home Component - Wrapper component for all elements on the blog page
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";
import { Helmet } from "react-helmet";

// Local JS Imports
import { getBlogPostList } from './../actions/actions';
import { BlogList } from './components/blog_list';

// CSS Imports 
import './blog.css';




// ===========
// Blog Home Component 
// ===========


export class BlogHomeWrapper extends Component {


	componentDidMount() {

		this.props.getBlogPostList();
	}


	render() {


		let result = (


				<div className="blog-page-wrapper">

					<Helmet>
				
						<title>Pundits Review Blog</title>
						<meta property="og:description" content="The Latest Stories from Pundits' Review"/>
						<meta property="og:url" content="https://www.punditsreview.com/blog"/>
			    		<meta property="og:type" content="football blog data"/>

			    	</Helmet>

					<p className="heading-text blog-page-title">Pundits Review Blog</p>

					<p className="body-text-light blog-page-description">LATEST STORIES</p>

					<BlogList />

				</div>


			);


		return result;
	}
}





// ===========
// Connect component to redux store
// ===========
const mapStateToProps = state => {
  return { 

  		blogPostList: state.blogPostList

  	};
};


export const BlogHome = connect(mapStateToProps, {getBlogPostList})(BlogHomeWrapper);
