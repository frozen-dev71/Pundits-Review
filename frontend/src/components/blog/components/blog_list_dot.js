// ======================================
// Blog List Dot Component - Displays a small dot
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";


// CSS Imports 
import './blog_list_dot.css';



// ===========
// Blog List Dot Component 
// ===========


export class BlogListDot extends Component {


	render() {

		let result = <span className="blog-list-dot"></span>


		return result;
	}
}


