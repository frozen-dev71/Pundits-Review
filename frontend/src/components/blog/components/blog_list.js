// ======================================
// Blog List Component - Will display a list of the currently available blog posts
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";
import { usePromiseTracker } from "react-promise-tracker";

// Local JS Imports 
import { Loading } from '../../global/components/loading';
import { BlogListItem } from './blog_list_item';

// CSS Imports
import './blog_list.css';


// ===========
// Blog List Component  
// ===========

export const BlogListWrapper = props => {


	// Display Loading screen until promise has resolved
	const { promiseInProgress } = usePromiseTracker();

	let result;

	if (promiseInProgress === false) {

		result = (

			<div className="blog-list-wrapper">

				{props.blogPostList.map((item, index) => {

					return <BlogListItem key={index} imageURL={item.header_image_url} headline={item.headline} date={item.date_text} description={item.thumbs_par} postName={item.post_name}/>


				})}

			</div>

			);


	} else {

		result = (

				<div className="blog-list-loading-container">

					<Loading />

				</div>

			);
	}


	return result;



}



// ===========
// Connect component to redux store
// ===========
const mapStateToProps = state => {
  return { 

  		blogPostList: state.blogPostList

  	};
};


export const BlogList = connect(mapStateToProps)(BlogListWrapper);
