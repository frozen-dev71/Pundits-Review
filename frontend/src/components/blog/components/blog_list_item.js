// ======================================
// Blog List Item Component - Displays One blog item - headline, image, & description
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { Link } from 'react-router-dom';

// Local JS Imports
import { BlogListDot } from './blog_list_dot';

// CSS Imports 
import './blog_list_item.css';


// ===========
// Blog List Item Component 
// ===========


export class BlogListItem extends Component {


	render() {

		let itemStyle = {

			backgroundImage: "url(" + this.props.imageURL + ")",
		}

		let blogPostAddress = "/blogpost/" + this.props.postName; 

		let result = (

				<Link to={blogPostAddress} className="react-link">

					<div className="blog-list-item-wrapper">

						<span className="blog-list-item-image" style={itemStyle}></span>

						<p className="body-text-light blog-list-item-date">{this.props.date}</p>

						<p className="body-text-light blog-list-item-headline">{this.props.headline}</p>

						<BlogListDot />

						<p className="body-text-light blog-list-item-description">{this.props.description}</p>

					</div>

				</Link>

			);


		return result;
	}
}