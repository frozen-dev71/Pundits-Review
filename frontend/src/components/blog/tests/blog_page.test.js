// ======================================
// This file tests for the components listed in the blog sub-directory
// ======================================


// ===========
// TEST COVERAGE
// ===========

// 1. Check that BlogHome renders correctly
// 2. Check that BlogHome has 2 p elements
// 3. Check that BlogList renders correctly
// 4. Check that BlogList renders no blog posts without API call
// 5. Check that BlogListItem renders correctly

// ===========
// IMPORTS 
// ===========


// General Imports
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import toJson from 'enzyme-to-json';

// Local JS Imports
import { BlogHome } from '.././blog';
import { BlogList } from '.././components/blog_list';
import { BlogListItem } from '.././components/blog_list_item';

import { getBlogPostList } from '../../actions/actions';
import store from '../../store/store';
import { WrapStore } from '../../global/functions/test_helpers';


// Configurations
Enzyme.configure({ adapter: new Adapter() })


// ===========
// Suppression of SSL Warning 
//
// Code adapted from comment from sketchbuch on GitHub issues post 'Warning: useLayoutEffect does nothing on the server #1373' on 12 Oct 2019
// https://github.com/reduxjs/react-redux/issues/1373
// Accessed 09 Sep 2020
//
// __Note__:
//	
//		- Jest produces "Warning: useLayoutEffect does nothing on the server" - warning which confuses test cases 
// 		- Wrapper test suppresses that warning for clear test reports
// ===========

describe('Blog Tests', () => {
  	const originalConsoleError = console.error;

  	beforeEach(() => {
    	console.error = jest.fn((msg) => {
	      	if (msg.includes('Warning: useLayoutEffect does nothing on the server')) {
	       		return null;
	      	} else {
	        	originalConsoleError(msg);
	      	}
    	});
  	});

  afterEach(() => {
    console.error = originalConsoleError;
  });




  	// ===========
	// TEST 1. Check that BlogHome renders correctly
	// ===========

	describe('BlogHome', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<Provider store={store}><BlogHome/></Provider>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})




	// ===========
	// TEST 2. Check that BlogHome has 2 p elements
	// ===========

	describe('BlogHome', () => {

	  	it('should render 2 p elements', () => {

		  	// Declare component
		  	const component = (<Provider store={store}><BlogHome/></Provider>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    const paragraphs = wrapper.find('p');
		    expect(paragraphs).toHaveLength(2);
		    expect(paragraphs).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 3. Check that BlogList renders correctly
	// ===========

	describe('BlogList', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<Provider store={store}><BlogList/></Provider>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 4. Check that BlogList renders no blog posts without API call
	// ===========

	describe('BlogList', () => {

	  	it('should render 0 Blog Posts without API Call', () => {

		  	// Declare component
		  	const component = (<Provider store={store}><BlogList/></Provider>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    const blog_list_items = wrapper.find(BlogListItem);
		    expect(blog_list_items).toHaveLength(0);

	  	});

	})


	// ===========
	// TEST 5. Check that BlogListItem renders correctly
	// ===========

	describe('BlogListItem', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BlogListItem/>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})

	


});
