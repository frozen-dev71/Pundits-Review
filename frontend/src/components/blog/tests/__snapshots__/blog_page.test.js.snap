// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Tests BlogHome should render 2 p elements 1`] = `
Array [
  <p
    class="heading-text blog-page-title"
  >
    Pundits Review Blog
  </p>,
  <p
    class="body-text-light blog-page-description"
  >
    LATEST STORIES
  </p>,
]
`;

exports[`Blog Tests BlogHome should render correctly 1`] = `
<div
  class="blog-page-wrapper"
>
  <p
    class="heading-text blog-page-title"
  >
    Pundits Review Blog
  </p>
  <p
    class="body-text-light blog-page-description"
  >
    LATEST STORIES
  </p>
  <div
    class="blog-list-wrapper"
  />
</div>
`;

exports[`Blog Tests BlogList should render correctly 1`] = `
<div
  class="blog-list-wrapper"
/>
`;

exports[`Blog Tests BlogListItem should render correctly 1`] = `
<Link
  className="react-link"
  to="/blogpost/undefined"
>
  <div
    className="blog-list-item-wrapper"
  >
    <span
      className="blog-list-item-image"
      style={
        Object {
          "backgroundImage": "url(undefined)",
        }
      }
    />
    <p
      className="body-text-light blog-list-item-date"
    />
    <p
      className="body-text-light blog-list-item-headline"
    />
    <BlogListDot />
    <p
      className="body-text-light blog-list-item-description"
    />
  </div>
</Link>
`;
